src/generators/angular/AngularServicesMethodGenerator.ts

```diff
@@ -31,32 +32,42 @@ export class AngularServicesMethodGenerator {
             (methodModel): OptionalKind<MethodDeclarationStructure> => ({
                 scope: Scope.Public,
                 name: methodModel.name,
-                parameters: this.getMethodParameters(methodModel).map((p) => this.getParameterStatement(p)),
+                parameters: this.getMethodParameters(methodModel),
                 returnType: this.getMethodReturnType(methodModel),
                 statements: (writer) => {
+                    const options = this.settings.withRequestOptions
+                        ? { name: 'options' }
+                        : undefined;
+
                     if (methodModel.kind === MethodKind.Download) {
-                        this.createDownloadMethod(writer, methodModel);
+                        this.createDownloadMethod(writer, methodModel, options);
                     } else {
-                        this.createMethod(writer, methodModel);
+                        this.createMethod(writer, methodModel, options);
                     }
                 }
             })
         );
     }
 
-    protected getMethodParameters(methodModel: IMethodModel): IMethodParameter[] {
+    protected getMethodParameters(methodModel: IMethodModel): OptionalKind<ParameterDeclarationStructure>[] {
+        const statements = methodModel.parameters.map(x => this.getParameterStatement(x));
+
         if (this.settings.withRequestOptions) {
-            methodModel.parameters.push({
-                name: 'options',
-                place: ParameterPlace.Body,
-                optional: true,
-                dtoType: HTTP_REQUEST_OPTIONS,
-                isCollection: false,
-                isModel: false
-            });
+            statements.push(this.getRequestOptionsParameter());
         }
 
-        return methodModel.parameters;
+        return statements;
+    }
+
+    private getRequestOptionsParameter(): OptionalKind<ParameterDeclarationStructure> {
+        return this.getParameterStatement({
+            name: 'options',
+            place: ParameterPlace.Body,
+            optional: true,
+            dtoType: HTTP_REQUEST_OPTIONS,
+            isCollection: false,
+            isModel: false
+        });
```

Comment:
Maybe we could generate ParameterDeclarationStructure with method getParameterStatement?

There are alot of useless information like place, dtoType, isCollection, isModel